### PR TITLE
app-emulation/lxd:  Fix postinst for 0.16

### PR DESCRIPTION
**Pull request by erikmack**
Had exit in a function but I meant return.  Also, ensure that failure to add the bridge doesn't fail the build.  Also, patch out some sandbox-violating 'go get' invocations.

### CODE REVIEW

File: app-emulation/lxd/lxd-0.16.ebuild [hunk 1]
```diff
@@ -140,16 +142,32 @@ src_install() {
 }
 
 pkg_config() {
+
```
Comment (alunduil) [inline app-emulation/lxd/lxd-0.16.ebuild]: I'd say drop this extra blank line.

File: app-emulation/lxd/lxd-0.16.ebuild [hunk 2]
```diff
@@ -140,16 +142,32 @@ src_install() {
...
+	# Installing a bridge is a convenience for the user, but the ebuild
+	# should succeed even if this fails.  Hence all the error checking
+	# before an unconditional 0 return code.
 	if use daemon; then
-		if brctl show lxcbr0 2>&1 | grep "No such device" >/dev/null; then
-			brctl addbr lxcbr0
+		showoutput=$(brctl show lxcbr0 2>&1)
+		if test ${?} -eq 0 && test -n "${showoutput}"; then
+			if echo ${showoutput} | grep "No such device" >/dev/null; then
```
Comment (alunduil) [inline app-emulation/lxd/lxd-0.16.ebuild]: Would finding the index of this and testing on that be better than using echo and grep?  echo and grep seems out of place here.

**Comment by alunduil** [comment]
Can we look at reducing the cyclomatic complexity of the config function?  Is there a way to codify it as a switch or simpler cases that aren't so dependent?

**Comment by erikmack** [comment]
> Can we look at reducing the cyclomatic complexity of the config function?  Is there a way to codify it as a switch or simpler cases that aren't so dependent?

Yes, I'm going to remove the feature entirely.  It adds a temporary
bridge; the user still has to configure it manually to be permanent.

The reason for the elaborate error checking is that configuring
interfaces is well outside the scope of what ebuilds should be doing.

PR coming shortly.
-- 
Erik Mackdanz

**Pull request merged**